src/service-worker.js

```diff
@@ -1,6 +1,74 @@
 /* eslint-env serviceworker */
+import doRequest from './do-request';
+import DeviceStateStore from './device-state-store';
+
 self.PusherPushNotifications = {
+  endpointOverride: null,
   onNotificationReceived: null,
+
+  _endpoint: instanceId =>
+    self.PusherPushNotifications.endpointOverride
+      ? self.PusherPushNotifications.endpointOverride
+      : `https://${instanceId}.pushnotifications.pusher.com`,
+
+  _getVisibleClient: () =>
+    self.clients
+      .matchAll({
+        type: 'window',
+        includeUncontrolled: true,
+      })
+      .then(clients => clients.find(c => c.visibilityState === 'visible')),
```

Comment:
Why are we checking for visibility rather than focus?

Reply:
Seemed more in the spirit of the property being sent - doesn't really matter though as these properties can't be used to derive "openability" as we do for iOS/Android